SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -77,35 +77,73 @@ class const_ref {
   static std::string get_description() { return "const reference"; }
 
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems> call_simd_ctor(const DataT *ref_data) {
+  static void call_simd_ctor(const DataT *ref_data, DataT *out) {
     simd<DataT, NumElems> source_simd;
     source_simd.copy_from(ref_data);
-    return call_simd_by_const_ref<DataT, NumElems>(
-        simd<DataT, NumElems>(source_simd));
+    call_simd_by_const_ref<DataT, NumElems>(simd<DataT, NumElems>(source_simd),
+                                            out);
   }
 
 private:
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems>
-  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref) {
-    return simd_by_const_ref;
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
   }
 };
 
-int main(int argc, char **argv) {
-  sycl::queue queue{esimd_test::ESIMDSelector{},
-                    esimd_test::createExceptionHandler()};
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  static bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+                     const std::string &data_type) {
+    bool passed = true;
+
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>{queue});
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         shared_allocator<DataT>{queue});
```

Comment:
```suggestion
    shared_allocator<DataT> allocator(queue);
    shared_vector<DataT> result(NumElems, allocator);
    shared_vector<DataT> reference(ref_data, allocator);
```
What about something like this?

Reply:
When I try construct the vector by this state: ` shared_vector<DataT> shared_ref_data(ref_data, allocator);` I got this error:

```
ctor_copy.cpp:105:26: error: no matching constructor for initialization of 'shared_vector<unsigned long long>' (aka 'vector<unsigned long long, usm_allocator<unsigned
      long long, sycl::usm::alloc::shared>>')
    shared_vector<DataT> shared_ref_data(ref_data, allocator);
                         ^               ~~~~~~~~~~~~~~~~~~~
```
To apply your comment `ref_data` should be `shared_vector` type, but it just `std::vector` type

I've applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/614/commits/e87ad2ec4b8cff0070719ae8992d50d244c2a019

Reply:
Yep, sorry. It's a bit of surprise we cannot just state the source container with the only difference in allocator type.